### PR TITLE
feature (ref 17195): create ParagraphFactory.php to delete paragrafs …

### PR DESCRIPTION
**Pull request by jalalifardemos**
### Ticket : https://www.dev.diplanung.de/DefaultCollection/BOP-HH/_workitems/edit/17195

Description: create ParagraphFactory.php to delete paragrafs and new method in SingleDocumentHandler.php to delete single doc 

### Linked PRs (optional)
List other PRs that are somehow connected to this and explain the connection. Don't
link private repositories in public ones, but the other way around is fine.

- Addon: https://github.com/demos-europe/demosplan-addon/pull/116
- Cockpit: https://github.com/demos-europe/demosplan-addon-xbauleitplanung/pull/34

Delete the checkbox if it doesn't apply/isn't necessary.

- [ ] Tests updated/created
- [ ] Update documentation
- [x] Link all relevant tickets
- [x] Move the tickets on the board accordingly
- [ ] Data-Cy attributes added/updated ([conventions](https://dplan-documentation.demos-europe.eu/development/guidelines-conventions/coding-styleguides/twig_html.html#guideline-for-naming-cypress-hooks))

### CODE REVIEW

File: demosplan/DemosPlanCoreBundle/Logic/Factory/ParagraphFactory.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?php
+
```
Comment (MoritzMandler) [inline demosplan/DemosPlanCoreBundle/Logic/Factory/ParagraphFactory.php:2]: I would vote to put this method into the existing ```demosplan/DemosPlanCoreBundle/Logic/Document/ParagraphHandler``` and name it like the one above - like ```administrationDocumentDeleteHandler```

Then implement a ```ParagraphHandlerInterface``` and put that into demosplan-addon

**Review by MoritzMandler — CHANGES_REQUESTED** [review]

File: demosplan/DemosPlanCoreBundle/Logic/Factory/ParagraphFactory.php (hunk 1 above)
Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/Logic/Factory/ParagraphFactory.php]: done in [3fe7b86](https://github.com/demos-europe/demosplan-core/pull/3290/commits/3fe7b8642fce0c0ba7708864973c587d83b6be55)

File: demosplan/DemosPlanCoreBundle/Logic/Factory/ParagraphFactory.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+namespace demosplan\DemosPlanCoreBundle\Logic\Factory;
+
+use DemosEurope\DemosplanAddon\Contracts\Entities\ElementsInterface;
+use DemosEurope\DemosplanAddon\Contracts\Entities\ProcedureInterface;
+use DemosEurope\DemosplanAddon\Contracts\Factory\ParagraphFactoryInterface;
+use demosplan\DemosPlanCoreBundle\Repository\ParagraphRepository;
+
+class ParagraphFactory implements ParagraphFactoryInterface
```
[the remark this answers is not in the document]
Comment (MoritzMandler) [reply demosplan/DemosPlanCoreBundle/Logic/Factory/ParagraphFactory.php]: changed to ParagraphHandlerInterface

**Review by MoritzMandler — DISMISSED** [review]
looks good to me - thx

**Comment by MoritzMandler** [comment]
Bitte erst nach [dem Addon](https://github.com/demos-europe/demosplan-addon/pull/116) mergen.
Dort fehlt noch der changeLog.

File: composer.json [hunk 3]
```diff
@@ -33,7 +33,7 @@
         "cocur/slugify": "^4.0",
         "composer/package-versions-deprecated": "1.11.99.5",
         "demos-europe/edt-extra": "^0.24.0",
-        "demos-europe/demosplan-addon": "^0.24",
+        "demos-europe/demosplan-addon": "dev-ADO_t17195_sprint_cockpit_update",
```
Comment (breierDemos) [inline composer.json:36]: This should be adjusted to an new demosplan-addon version with the required changes.

**Review by breierDemos — CHANGES_REQUESTED** [review]
code review only

**Comment by jalalifardemos** [comment]
https://github.com/demos-europe/demosplan-core/pull/3534

**Pull request closed without merging**